internal/settings/ovh.go

```diff
@@ -142,3 +169,77 @@ func (o *ovh) Update(ctx context.Context, client *http.Client, ip net.IP) (newIP
 		return nil, fmt.Errorf("%w: %s", ErrUnknownResponse, s)
 	}
 }
+
+func (o *ovh) updateWithZoneDNS(client *ovhClient.Client, ip net.IP) (newIP net.IP, err error) {
+	recordType := A
+	var ipStr string
+	if ip.To4() == nil { // IPv6
+		recordType = AAAA
+		ipStr = ip.To16().String()
+	} else {
+		ipStr = ip.To4().String()
+	}
+	// subDomain filter of the ovh api expect an empty string to get @ record
+	subDomain := o.host
+	if subDomain == "@" {
+		subDomain = ""
+	}
+	// get existing records
+	var recordIDs []uint64
+	if err := client.Get(fmt.Sprintf("/domain/zone/%s/record?fieldType=%s&subDomain=%s", o.domain,
+		recordType, subDomain), &recordIDs); err != nil {
+		return ip, err
+	}
```

Comment:
I'm going to sound like a picky coder, but can you please use `client.Do` instead and create the request with `request := http.NewRequestWithContext(ctx, http.MethodGet, "url", nil)`? That way the request/client processing stops if the context gets canceled, I think in this case it only happens when the program gets interrupted but it could be important.

Same comment for the `client.Post`. It's a bit longer but there is no way around it unfortunately for now with the Go API. Note that for the Post, you can use

```go
buffer := bytes.NewBuffer(nil)
encoder := json.NewEncoder(buffer)
if err := json.Encode(yourstruct); err != nil {
     ....
}
```

to encode your json struct and then pass the buffer as the last argument of `http.NewRequestWithContext`

Reply:
client.Do is used under the hood https://github.com/ovh/go-ovh/blob/302dfc012714871f55e4ce3bae69f903fe28327b/ovh/ovh.go#L387

Reply:
Oh my bad, I thought `client` was an `*http.Client`, glad you found these methods too! Perfect!